clippy_lints/src/matches.rs

```diff
@@ -311,6 +311,36 @@ declare_clippy_lint! {
     "a match with a single binding instead of using `let` statement"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for unnecessary '..' pattern binding on struct when all fields are explicitly matched.
+    ///
+    /// **Why is this bad?** Correctness and readability. It's like having a wildcard pattern after
+    /// matching all enum variants explicitly which is forbidden.
```

Comment:
```suggestion
    /// matching all enum variants explicitly.
```

it's not  forbidden, only linted against by rustc.